### PR TITLE
Fix lint errors, refuse to build unlinted code.

### PR DESCRIPTION
**Pull request by alanconway**
/assign @jcantrill
/cc @periklis

### CODE REVIEW

File: pkg/k8shandler/defaults.go [hunk 1]
```diff
@@ -19,7 +19,4 @@ var (
 
 	defaultFluentdMemory     resource.Quantity = resource.MustParse("736Mi")
 	defaultFluentdCpuRequest resource.Quantity = resource.MustParse("100m")
-
```
Comment (vimalk78) [inline pkg/k8shandler/defaults.go:22]: unrelated change?

File: Makefile [hunk 2]
```diff
@@ -32,7 +32,7 @@ operator-sdk:
 golangci-lint:
 	@type -p golangci-lint > /dev/null || GOFLAGS="" GO111MODULE=off go get github.com/golangci/golangci-lint/cmd/golangci-lint
 
-build:
+build: lint		# Reject unlinted code in CI.
```
Comment (periklis) [inline Makefile:35]: Can we re-evaluate the set of linters and the golangci-lint version we use in a follow-up PR please? I tend to believe that we are using really old linters.

Comment (alanconway) [reply Makefile:35]: Sure, open a JIRA or just start a PR with proposal.

File: pkg/k8shandler/defaults.go (hunk 1 above)
Comment (alanconway) [reply pkg/k8shandler/defaults.go:22]: Nope, those are unused variables, they cause lint to fail. They are the reason I was pushed to do this - tired of getting other peoples lint in my PRs :)

**Comment by alanconway** [comment]
> Is this going to work with our CI building images
> 
> /lgtm cancel

@periklis That's the idea. I want CI to fail if someone's PR has linty code so they have fix it before it infects master.
I originally suggested changing the CI script, but @jcantrill felt it would be easier to just ad to build: target.

**Comment by alanconway** [comment]
Looks like CI is faliing to `go get` the golangci-lint tool:

**Comment by alanconway** [comment]
@periklis @igor-karpukhin rebased and improved a little, needs LGTM

**Comment by periklis** [comment]
> CI was failing in `go get`. I switched to the module version of `go get` since gloagnci-lint now supports that.
> It required some module changes and a surprising number of vendor changes - @periklis can you say if it makes sense?

Since golangci-lint is a go modules project, doing `go get` w/o prepending `GOFLAGS=""...` would result in adding this tool into our `go.mod` definitions and don't want this for now. At least as long as we don't have a conclusive decision how we want to include tools.
